products/storage/api.yaml

```diff
@@ -21,6 +21,35 @@ versions:
 scopes:
   - https://www.googleapis.com/auth/devstorage.full_control
 objects:
+  - !ruby/object:Api::Resource
+    name: 'Object'
+    base_url: b/{{bucket}}/o/
+    description: Upload or download a file from a GCS bucket.
+    exclude: true
+    parameters:
+      - !ruby/object:Api::Type::Enum
+        name: 'action'
+        description: 'Upload or download from the bucket'
+        values:
+          - :download
+          - :upload
+      # TODO(alexstephen): Figure out why MD5 hashes are different local vs. remote
```

Comment:
That sounds bad

Reply:
Yeah, that stumped me for the longest time. Turns out that I was doing everything the hard way. The `requests` library handles all of the file-upload logic for me (and it does it correctly!)